httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/HttpAsyncClientEventHandlerFactory.java

```diff
@@ -189,7 +189,7 @@ private void logFlowControl(final String prefix, final int streamId, final int d
                         public void onHeaderInput(final HttpConnection connection, final int streamId, final List<? extends Header> headers) {
                             if (headerLog.isDebugEnabled()) {
                                 for (int i = 0; i < headers.size(); i++) {
```

Comment:
This should be a foreach loop, index-based access to lists is discouraged: separate ticket.